src/Microsoft.AspNetCore.Antiforgery/IAntiforgery.cs

```diff
@@ -56,6 +57,16 @@ public interface IAntiforgery
         Task ValidateRequestAsync(HttpContext httpContext);
 
         /// <summary>
+        /// Validates an antiforgery token that was supplied as part of the request.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> associated with the current request.</param>
+        /// <param name="principal">The claims-based principal to validate.</param>
+        /// <exception cref="AntiforgeryValidationException">
+        /// Thrown when the request does not include a valid antiforgery token.
+        /// </exception>
+        Task ValidateRequestAsync(HttpContext httpContext, ClaimsPrincipal principal);
```

Comment:
@dougbu - should we add the same for `IsRequestValidAsync`? I don't need it for what I'm doing but it would be consistent.